src/frontend/src/flows/authenticate/index.ts

```diff
@@ -24,72 +26,106 @@ import {
 } from "./validateDerivationOrigin";
 import { unreachable } from "../../utils/utils";
 
+type PageElements = {
+  authorizeButton: HTMLButtonElement;
+  userNumberInput: HTMLInputElement;
+  registerButton: HTMLLinkElement;
+};
+
 const pageContent = (
+  connection: Connection,
   hostName: string,
   userNumber?: bigint,
   derivationOrigin?: string
-) => html` <div class="l-container c-card c-card--highlight">
-    <!-- The title is hidden but used for accessibility -->
-    <h1 class="is-hidden">Internet Identity</h1>
-    <div class="c-logo">${icLogo}</div>
-    <div class="l-stack">
-      <p class="t-lead" style="text-align: center;">
-        Connect to<br />
-        <span class="t-strong">${hostName}</span>
-        ${derivationOrigin !== undefined && derivationOrigin !== hostName
-          ? html` <span class="c-tooltip t-link__icon"
-              >${attentionIcon}<i
-                class="c-tooltip__message c-card c-card--narrow"
-                >An alternative origin of ${derivationOrigin}</i
-              ></span
-            >`
-          : ""}
-        <br />
-      </p>
-    </div>
+): TemplateRef<PageElements> => {
+  const onManageClick = () => {
+    window.location.hash = "";
+    window.location.reload();
+  };
 
-    <div class="l-stack c-input c-input--vip c-input--anchor">
-      <input
-        type="text"
-        id="userNumberInput"
-        placeholder="Enter anchor"
-        value="${userNumber !== undefined ? userNumber : ""}"
-        style="width: 100%;"
-      />
+  const onRecoverClick = () => useRecovery(connection, readUserNumber());
 
-      <p
-        id="invalidAnchorMessage"
-        class="anchor-error-message is-hidden t-paragraph t-strong"
-      >
-        The Identity Anchor is not valid. Please try again.
-      </p>
-    </div>
+  const authorizeButton = createRef();
```

Comment:
would it make sense to postfix the variables with the keyword `Ref`?

e.g. `authorizeButtonRef`. verbose but clear.

Reply:
I actually did this at first, but then on the `return` I was just renaming all keys (authorizeButtonRef -> authorizeButton) so I just dropped the suffix altogether

Reply:
rename also where you use it 🤷‍♂️

Reply:
no, because where it's used it's not a ref anymore; that's the translation `TemplateRef` does at the type level; alternatively I could use that step to rename the keys at type level as well but this is getting a little crazy